### PR TITLE
AQ-46 axios react query

### PR DESCRIPTION
**Pull request by ToJestKrzysio**
# Description
Added `axios` along side `@tanstack/react-query`.
Setup react-query within `App.tsx`.
Added `TestQuery` component to manually verify if everything works according to expectations.
Updated `vite.config.ts`, `tsconfig.json`, `.eslintrc.json` to work with absolute imports.
Updated `vite.config.ts` to prevent `ESlint` from breaking development on error.
Configured `Vitest` as test runner, added `testing-library` and`mock service worker` to support React testing.
Added tests for anythin i thought of.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The related memme has been added

image

### CODE REVIEW

File: quiz/docker-compose.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ services:
       - envs/web.env
     command: /bin/bash -c "npm ci && npm run dev"
     stdin_open: true
+    tty: true
```
Comment (ToJestKrzysio) [inline quiz/docker-compose.yaml:16]: Prettier logs

File: quiz/web/vite.config.ts [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
 export default defineConfig({
     server: {
         port: parseInt(process.env.VITE_PORT),
         host: "0.0.0.0",
     },
-    plugins: [react(), eslint({ fix: true })],
+    plugins: [
+        react(),
+        tsconfigPaths(),
+        { ...eslint({ failOnError: true, failOnWarning: false }), apply: "build" },
```
Comment (ToJestKrzysio) [inline quiz/web/vite.config.ts:14]: Updated `eslint` configuration for Vite so it does not break development on error.

**Review by olivergav — APPROVED** [review]

**Review by olivergav — APPROVED** [review]

File: quiz/web/package.json [hunk 3]
```diff
@@ -7,13 +7,25 @@
         "dev": "vite",
         "build": "tsc && vite build",
         "preview": "vite preview",
-        "lint": "prettier --write ."
+        "lint": "prettier --write .",
```
Comment (pawelkonior) [inline quiz/web/package.json:10]: chcemy, żeby prettier tyrał po node modules? \"src/**/*.{ts,tsx,js,jsx}\

**Review by pawelkonior — APPROVED** [review]
Wygląda pieknie

File: quiz/web/package.json (hunk 3 above)
Comment (ToJestKrzysio) [reply quiz/web/package.json]: Fixed

**Pull request merged**